components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -5854,7 +5854,9 @@ def getBody(self):
             if self.descriptor.hasLegacyUnforgeableMembers:
                 raise TypeError("Can't handle a deleter on an interface that has "
                                 "unforgeables. Figure out how that should work!")
-            set += CGProxyNamedDeleter(self.descriptor).define()
+            set += ("if id.is_string() || id.is_int() {\n"
```

Comment:
This isn't quite right. https://searchfox.org/mozilla-central/source/__GENERATED__/dom/bindings/StorageBinding.cpp#1221 is the model we should use, which 1. Verifies that a symbol is present, 2) does not make any function calls when that's the case, 3) returns an error

Reply:
I think we'll need to modify the logic of CGProxyNamedDeleter instead.

Reply:
Done